subsys/net/ip/net_context.c

```diff
@@ -229,7 +229,9 @@ int net_context_get(sa_family_t family,
 			continue;
 		}
 
-		if (ip_proto == IPPROTO_TCP) {
+		if (!(IS_ENABLED(CONFIG_NET_OFFLOAD) &&
```

Comment:
I personally can't parse these conditions by just looking at them. Can we please make net_if_is_ip_offloaded() just return const `false` for the case of !CONFIG_NET_OFFLOAD ?

Reply:
> I personally can't parse these conditions by just looking at them. Can we please make net_if_is_ip_offloaded() just return const `false` for the case of !CONFIG_NET_OFFLOAD ?

I aslo don't like this style, But there are lots of similar places in net_context.c,  i,e. "IS_ENABLED(CONFIG_NET_OFFLOAD) && net_if_is_ip_offloaded(net_context_get_iface(context)".  What's more, net_if_is_ip_offloadded can return false for the case of !CONFIG_NET_OFFLOAD.  

So it looks that we need to a clean up in net_context.c., better have the maintainer of net context.c to do that

In this PR,  i will fix my part.

Reply:
> But there are lots of similar places in net_context.c ...
> better have the maintainer of net context.c to do that...

True, and @jukkar and @tbursztyka regularly do rounds on cleanups on that. From our side, we, as contributors, can help them at least in small places which we touch while working on other code.

> In this PR, i will fix my part.

Thanks!

Reply:
There is some discussion of having L2 offload support in #18117, it will most probably affect the offload checks in the code. Anyway, we will see when we get there as no PR has yet been submitted.

Reply:
This line fixed.